doc/TESTING_YOUR_CHANGES.md

```diff
@@ -12,6 +12,8 @@ so read this one first.
 
 For JSON changes, read [JSON_STYLE.md](JSON_STYLE.md).
 For C++ changes, read [CODE_STYLE.md](CODE_STYLE.md).
+If tinkering with C++, you may also want to look at
+[TESTING.md](TESTING.md) for our unit tests.
```

Comment:
Please move this to the "applying your changes locally" section, after the mention of `COMPILING.md`.

> I am unfamiliar with running tests manually. Why and when would you do that?

When you change the unit test code itself, change C++ or JSON code covered by the test cases, or need to fix failing CI unit tests caused by your PR.

It is not uncommon for people to submit their changes without running the unit test though, and the CI unit test should catch any problems in that case, so a simple mention that the unit test exists should be enough.